src/utils/credentials-helper.js

```diff
@@ -0,0 +1,244 @@
+    .addStringNoLocale(RDF_PREDICATES.sha256, publicKey)
+    .build();
+  const publicKeyDataSet = setThing(createSolidDataset(), publicKeyThing);
+
+  return {
+    private: privateKeyDataSet,
+    public: publicKeyDataSet
+  };
+};
+
+const savePassUserCredentials = async (dataSets, session) => {
+  const privateKeyDataSet = dataSets.private;
+  const publicKeyDataSet = dataSets.public;
+  const containerUrl = `${String(session.info.webId.split('profile')[0])}PASS_Credentials/`;
+  const privateKeyUrl = `${String(
+    session.info.webId.split('profile')[0]
+  )}PASS_Credentials/private_key.ttl`;
+  const publicKeyUrl = `${String(
+    session.info.webId.split('profile')[0]
+  )}PASS_Credentials/public_key.ttl`;
+
+  await saveSolidDatasetAt(privateKeyUrl, privateKeyDataSet, {
+    fetch: session.fetch
+  });
+  await saveSolidDatasetAt(publicKeyUrl, publicKeyDataSet, {
+    fetch: session.fetch
+  });
+
+  const keyAccessAcl = {
+    read: true,
+    write: false,
+    append: false,
+    control: true
+  };
+
+  await createDocAclForUser(session, containerUrl, keyAccessAcl);
+  await createDocAclForUser(session, privateKeyUrl, keyAccessAcl);
+  await createDocAclForUser(session, publicKeyUrl, keyAccessAcl);
+};
+
+const getUserSigningKeyInternal = async (session) => {
+  const privateKeyUrl = `${String(
+    session.info.webId.split('profile')[0]
+  )}PASS_Credentials/private_key.ttl`;
+  const dataSet = await getSolidDataset(privateKeyUrl, { fetch: session.fetch });
+  const keyString = getStringNoLocale(
+    getThing(dataSet, `${privateKeyUrl}#privateKey`),
+    RDF_PREDICATES.sha256
+  );
+  const key = await window.crypto.subtle.importKey(
+    'jwk',
+    JSON.parse(keyString),
+    {
+      name: 'RSA-PSS',
+      hash: 'SHA-256'
+    },
+    false,
+    ['sign']
+  );
+  return key;
+};
+
+const InitializePassUserCredentials = async (session) => {
+  const dataSets = await createPassUserCredentials(session);
+  return savePassUserCredentials(dataSets, session);
+};
+
+/**
+ * Fetches user signing key from user's pod. If one does not exist, creates one, then fetches it.
+ *
+ * @memberof utils
+ * @function getUserSigningKey
+ * @param {Session} session - Solid's Session Object (see {@link Session})
+ * @returns {Promise} Promise - File upload is handled via Solid libraries
+ */
+export const getUserSigningKey = async (session) => {
+  let key = null;
+  try {
+    try {
+      key = await getUserSigningKeyInternal(session);
+    } catch (e) {
+      await InitializePassUserCredentials(session);
+      key = await getUserSigningKeyInternal(session);
+    }
+  } catch (e) {
+    key = null;
+  }
+  return key;
+};
+
+/**
+ * Serializes a given Solid DataSet into TTL
+ * Note: Should not be used for networking. Use inrupt's libraries for that
+ *
+ * @memberof utils
+ * @function serializeDataSet
+ * @param {import('@inrupt/solid-client').SolidDataset} dataSet - Inrupt Dataset you want to serialize
```

Comment:
For importing types, we could use `@typedef` from JSDoc (see Lines 35 to 45 in session-core.js).

After that you could use SolidDataset as a type using just `{SolidDataset}` in JSDoc.

The type `{Session}` is a good example of this in Line 58 of session-core, the imported `@typedef` could be found in Lines 35 to 37.